h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -444,47 +481,54 @@ public static AstNumList check(long dstX, AstRoot ast) {
   // the selected 'gbCols' columns, and for each group compute aggregrate
   // results using 'aggs'.  Return an array of groups, with the aggregate results.
   public static IcedHashSet<G> doGroups(Frame fr, int[] gbCols, AGG[] aggs) {
-    return doGroups(fr, gbCols, aggs, false, true);
+    final byte[] frameColsTypes = fr.types();
+    final int[] gbColsTypes = Arrays.stream(gbCols).map(gbColIdx -> frameColsTypes[gbColIdx]).toArray();
```

Comment:
duplicated code, maybe move to ArrayUtils

Reply:
Good point. I implemented `ArrayUtils.select` for `byte[]` and used `byte[]` throughout the code.